tasks/libs/junit_upload.py

```diff
@@ -0,0 +1,134 @@
+import glob
+import io
+import os
+import platform
+import subprocess
+import tarfile
+import tempfile
+import xml.etree.ElementTree as ET
+
+CODEOWNERS_ORG_PREFIX = "@DataDog/"
+REPO_NAME_PREFIX = "github.com/DataDog/datadog-agent/"
+DATADOG_CI_COMMAND = ["datadog-ci", "junit", "upload"]
+JOB_URL_FILE_NAME = "job_url.txt"
+TAGS_FILE_NAME = "tags.txt"
+
+
+def split_junitxml(xml_path, codeowners, output_dir):
+    """
+    Split a junit XML into several according to the suite name and the codeowners.
+    Returns a list with the owners of the written files.
+    """
+    tree = ET.parse(xml_path)
+    output_xmls = {}
+    for suite in tree.iter("testsuite"):
+        path = suite.attrib["name"].replace(REPO_NAME_PREFIX, "", 1)
+
+        owners = codeowners.of(path)
+        if not owners:
+            raise KeyError("No code owner found for {}".format(path))
+
+        main_owner = owners[0][1][len(CODEOWNERS_ORG_PREFIX) :]
```

Comment:
If we're tagging only by the first codeowner, we may want to rework some lines in the `CODEOWNERS` file, eg. https://github.com/DataDog/datadog-agent/blob/b377f242267dd3fe361cb6f99cbcc344144f518e/.github/CODEOWNERS#L159-L160, https://github.com/DataDog/datadog-agent/blob/b377f242267dd3fe361cb6f99cbcc344144f518e/.github/CODEOWNERS#L83, etc. where the "first" owner isn't necessarily the main owner.

Reply:
Yeah, but this should really be decided by the individuals/teams that work on these, so I don't want to mix this in the PR. I'll send the communication about this when this is merged, so people can update codeowners as they want.